crates/fluvio/src/consumer/retry.rs

```diff
@@ -175,33 +197,31 @@ impl Stream for ConsumerRetryStream {
 
 impl ConsumerStream for ConsumerRetryStream {
     fn offset_commit(&mut self) -> ConsumerBoxFuture {
-        let notify = self.notify.clone();
         Box::pin(async move {
-            loop {
-                match self.state {
-                    ConsumerRetryState::Idle | ConsumerRetryState::Terminated => {
-                        return self.stream.lock().await.offset_commit().await;
-                    }
-                    _ => {
-                        notify.notified().await;
-                    }
-                }
+            if let Some(mut stream) = self.stream.try_lock() {
```

Comment:
Seems like you could put timer here which would avoid need to put another layer on top.
So you could do:

```
        let timer = fluvio::sleep(timer).
         loop {
              select! {
              
                   etc

            }
         
         
         }
```

Reply:
A loop here would not work well in the case where stream is locked waiting for a record that never arrives.

But I agree that this code smells, thinking in somehow to improve it. 

I removed the expected for now.

Reply:
maybe if we have a timer also in `handle_consumer_records`, to not allow lock it forever. So we will not need this layer

Reply:
I removed this layer